### ISSUE TITLE
[Data Grid] Page size does not reset after clearing a filter when using `autoPageSize`

### ISSUE THREAD

**Issue by bahador-worklete** [mui-org/material-ui-x#1413]
When using `autoPageSize` with a `DataGrid`, if you set a filter and clear it, the page size does not reset.

- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/mui-org/material-ui-x/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

Let's say I have a DataGrid using autoPageSize with 100 rows, a page size of 20, and 5 pages. If I set a filter that returns 1 row, I will have 1 page with 1 row. If I clear that filter, I now have 100 pages with a page size of 1.
## Expected Behavior 🤔

After clearing the filter, I would expect to have 5 pages of 20, just as I started out with when loading the DataGrid.

## Steps to Reproduce 🕹

Steps:

1. Go to this link: https://material-ui.com/components/data-grid/pagination/#auto-size
2. Create a filter on the `Trader Name` field using any one of the populated values for that field
3. Notice that now there is only a single result and that the page size is 1
4. Clear the filter, and now you will see that the page size remains 1 but that you have 100 pages

## Context 🔦

I just want the Data Grid to return to the same size it was when it loaded.

## Your Environment 🌎

<details>
  <summary>`npx @material-ui/envinfo`</summary>
  
Chrome Version 89.0.4389.114 (Official Build) (x86_64)
```
npx: installed 2 in 3.204s

  System:
    OS: macOS 10.15.7
  Binaries:
    Node: 12.21.0 - ~/.nvm/versions/node/v12.21.0/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 6.14.11 - ~/.nvm/versions/node/v12.21.0/bin/npm
  Browsers:
    Chrome: 89.0.4389.114
    Edge: Not Found
    Firefox: 87.0
    Safari: 13.1.3
  npmPackages:
    @material-ui/core: 4.11.0 => 4.11.0 
    @material-ui/data-grid: ^4.0.0-alpha.23 => 4.0.0-alpha.24 
    @material-ui/icons: ^4.9.1 => 4.9.1 
    @material-ui/lab: ^4.0.0-alpha.56 => 4.0.0-alpha.56 
    @material-ui/styles:  4.10.0 
    @material-ui/system:  4.9.14 
    @material-ui/types:  5.1.0 
    @material-ui/utils:  4.10.2 
    @types/react: ^16.9.44 => 16.9.56 
    react: 16.13.1 => 16.13.1 
    react-dom: 16.13.1 => 16.13.1 
    typescript: 3.9.7 => 3.9.7 
```
</details>

**Comment by oliviertassinari** [comment]
Can somebody test that it's the same root cause as #1334? Meaning that the fix we did cover this case too?

**Comment by dtassone** [comment]
Yes I checked it does cover it

**Comment by bahador-worklete** [comment]
I tested it on material-ui.com and it's broken there. I'm using `4.0.0-alpha.23` for my project and it's broken there.
Perhaps I'm doing something wrong?

**Comment by dtassone** [comment]
It will be fixed in the next release.

**Comment by bahador-worklete** [comment]
Thank you.

**Comment by oliviertassinari** [comment]
Fix released in v4.0.0-alpha.25

**Comment by bahador-worklete** [comment]
tyvm
